src/pyEQL/solution.py

```
@@ -3228,3 +3230,79 @@ def get_salt_list(self):  # pragma: no cover
+              Valid entries are 'seawater', 'rainwater', 'wastewater',
+              'urine', 'normal saline' and 'Ringers lactate'.
+
+        Returns:
+            A pyEQL Solution object.
+        """
+        # Path to the YAML and JSON files corresponding to the preset
+        yaml_path = os.path.join("presets", f"{preset}.yaml")
+        json_path = os.path.join("presets", f"{preset}.json")
+
+        # Check if the file exists
+        if os.path.exists(yaml_path):
+            preset_path = yaml_path
+        elif os.path.exists(json_path):
+            preset_path = json_path
+        else:
+            logger.error("Invalid solution entered - %s" % preset)
+            raise FileNotFoundError(f"Files '{yaml_path}' and '{json_path} not found!")
+
+        # Create and return a Solution object
+        return cls().from_file(preset_path)
+
+    def to_file(self, file_name: str, extension: str = "yaml") -> None:
+        """Saving to a .yaml or .json file.
+
+        Args:
+            file_name (str): The name of the file with the saved Solution
+            extension (str, optional): The extension of the save file.
+              Valid entries are 'yaml' and 'json'.
+              Defaults to 'yaml'.
+        """
+        if extension not in ["json", "yaml"]:
+            logger.error("Invalid extension entered - %s" % extension)
+            return
+        file_path = os.path.join("presets", f"{file_name}.{extension}")
+        if extension == "yaml":
+            solution_dict = self.as_dict()
+            dumpfn(solution_dict, file_path)
+        else:
+            dumpfn(self, file_path)
+
+    def from_file(self, file_path: str) -> Solution | None:
+        """Loading from a .yaml or .json file.
+
+        Args:
+            file_path (str): _description_
+        """
+        if not os.path.exists(file_path):
+            logger.error("Invalid path to file entered - %s" % file_path)
```

I think this should be a `raise FileNotFoundError` rather than a log message (similar to what you have in `from_preset`)